src/utils/collection-stream.js

```diff
@@ -0,0 +1,63 @@
+const stream = require('stream');
+const Writable = stream.Writable;
+const debug = require('./logger').createLogger('collection-stream');
+
+class WritableCollectionStream extends Writable {
+  constructor(dataService, ns) {
+    super({ objectMode: true});
+    this.dataService = dataService;
+    this.ns = ns;
+    this.BATCH_SIZE = 1000;
+    this.buf = [];
+    this.docsWritten = 0;
+  }
+
+  _write(chunk, encoding, next) {
+    this.buf.push(chunk);
+    if (this.buf.length === this.BATCH_SIZE) {
+      return this.dataService.insertMany(
+        this.ns,
+        this.buf,
+        { ordered: false },
+        (err) => {
+          this.docsWritten += this.buf.length;
+          this.buf = [];
+          if (err) {
+            debug('error', err);
+            return next(err);
+          }
+          next();
+        }
+      );
+    }
+    next();
```

Comment:
clarification: this is not a "termination" `next()`, right? this is meant to be "we are not at BATCH_SIZE, so don't flush" `next`?

Reply:
Correct!